pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/SegmentAbortedTxnProcessorTest.java

```diff
@@ -256,6 +261,59 @@ public void testClearSnapshotSegments() throws Exception {
         processor.closeAsync().get(5, TimeUnit.SECONDS);
     }
 
+    @Test
+    public void testTxnSegmentStats() throws Exception {
+        Transactions transactions = admin.transactions();
+        String PROCESSOR_TOPIC = "persistent://" + NAMESPACE1 + "/testTxnSegmentStats";
+        admin.topics().createNonPartitionedTopic(PROCESSOR_TOPIC);
+        // Prepare topic, producer, consumer
+        @Cleanup
+        Producer<byte[]> producer = pulsarClient.newProducer()
+                .topic(PROCESSOR_TOPIC)
+                .create();
+        @Cleanup
+        Consumer<byte[]> consumer = pulsarClient.newConsumer()
+                .topic(PROCESSOR_TOPIC)
+                .subscriptionName("my-sub")
+                .subscribe();
+
+        // Record the start time of the test
+        long testStartTime = System.currentTimeMillis();
+
+        // Send messages with transactions and abort them
+        for (int i = 0; i < SEGMENT_SIZE + 2; i++) {
+            Transaction transaction = pulsarClient.newTransaction()
+                    .withTransactionTimeout(5, TimeUnit.HOURS).build().get();
+            producer.newMessage(transaction).send();
+            transaction.abort();
+        }
+
+        // Get the transaction buffer stats without segment stats
+        TransactionBufferStats statsWithoutSegmentStats = transactions.getTransactionBufferStats(PROCESSOR_TOPIC,
+                false, false);
+        assertNotNull(statsWithoutSegmentStats);
+        assertNotNull(statsWithoutSegmentStats.segmentsStats);
+        assertNull(statsWithoutSegmentStats.segmentsStats.segmentStats);
+
+        // Get the transaction buffer stats with segment stats
+        TransactionBufferStats statsWithSegmentStats = transactions.getTransactionBufferStats(PROCESSOR_TOPIC,
+                false, true);
+        assertNotNull(statsWithSegmentStats);
+        assertNotNull(statsWithSegmentStats.segmentsStats);
+
+        // Verify the segment stats
+        assertEquals(statsWithSegmentStats.segmentsStats.segmentsSize, 1L);
+        assertEquals(statsWithSegmentStats.segmentsStats.unsealedAbortTxnIDSize, 2L);
+        assertEquals(statsWithSegmentStats.segmentsStats.currentSegmentCapacity, SEGMENT_SIZE);
+        assertTrue(statsWithSegmentStats.segmentsStats.lastTookSnapshotSegmentTimestamp >= testStartTime);
+
+        // Verify if the segment stats are present when requested
+        assertEquals(statsWithSegmentStats.segmentsStats.segmentStats.size(), 1);
+
+        // Verify the totalAbortedTransactions field
+        assertEquals(statsWithSegmentStats.totalAbortedTransactions, SEGMENT_SIZE + 2);
```

Comment:
please check `snapshotType`, and the attributes of `SegmentStats.lastTxnId` here.